aea/aea_builder.py

```diff
@@ -1228,46 +1255,34 @@ def _load_connection(
                 Connection,
                 self._component_instances[ComponentType.CONNECTION][configuration],
             )
-            if connection.address != address:
+            if connection.address != identity.address:
                 logger.warning(
                     "The address set on connection '{}' does not match the default address!".format(
                         str(connection.connection_id)
                     )
                 )
             return connection
-        try:
-            directory = cast(Path, configuration.directory)
-            package_modules = load_all_modules(
-                directory, glob="__init__.py", prefix=configuration.prefix_import_path
-            )
-            add_modules_to_sys_modules(package_modules)
-            connection_module_path = directory / "connection.py"
-            assert (
-                connection_module_path.exists() and connection_module_path.is_file()
-            ), "Connection module '{}' not found.".format(connection_module_path)
-            connection_module = load_module(
-                "connection_module", directory / "connection.py"
-            )
-            classes = inspect.getmembers(connection_module, inspect.isclass)
-            connection_class_name = cast(str, configuration.class_name)
-            connection_classes = list(
-                filter(lambda x: re.match(connection_class_name, x[0]), classes)
-            )
-            name_to_class = dict(connection_classes)
-            logger.debug("Processing connection {}".format(connection_class_name))
-            connection_class = name_to_class.get(connection_class_name, None)
-            assert (
-                connection_class is not None
-            ), "Connection class '{}' not found.".format(connection_class_name)
-            return connection_class.from_config(
-                address=address, configuration=configuration
+        else:
+            configuration = deepcopy(configuration)
+            return cast(
+                Connection,
+                load_component_from_config(
+                    ComponentType.CONNECTION,
+                    configuration,
+                    identity=identity,
+                    crypto_store=wallet.connection_cryptos,
```

Comment:
what about to filter connection cryptos by connection id?

Reply:
Perhaps in the future yes; as part of this release it won't happen.